tss-esapi/tests/context_tests/tpm_commands/symmetric_primitives_tests.rs

```diff
@@ -1,5 +1,146 @@
 // Copyright 2021 Contributors to the Parsec project.
 // SPDX-License-Identifier: Apache-2.0
+mod test_encrypt_decrypt_2 {
+    use crate::common::create_ctx_without_session;
+    use std::convert::TryFrom;
+    use tss_esapi::{
+        abstraction::cipher::Cipher,
+        attributes::ObjectAttributesBuilder,
+        constants::AlgorithmIdentifier,
+        interface_types::{
+            algorithm::SymmetricMode, resource_handles::Hierarchy, session_handles::AuthSession,
+        },
+        structures::{Auth, InitialValue, MaxBuffer},
+        utils::{PublicParmsUnion, Tpm2BPublicBuilder},
+    };
+    #[test]
+    fn test_encrypt_decrypt_with_aes_128_cfb_symmetric_key() {
+        let mut context = create_ctx_without_session();
+
+        context
+            .tr_set_auth(Hierarchy::Owner.into(), &Auth::default())
+            .expect("Failed to set auth to empty for owner");
+
+        let primary_key_auth = Auth::try_from(
+            context
+                .get_random(16)
+                .expect("get_rand call failed")
+                .value()
+                .to_vec(),
+        )
+        .expect("Failed to create primary key auth");
+
+        let primary_key_handle = context.execute_with_session(Some(AuthSession::Password), |ctx| {
+            ctx.create_primary(
+                Hierarchy::Owner,
+                &tss_esapi::utils::create_restricted_decryption_rsa_public(
+                    Cipher::aes_256_cfb(),
+                    2048,
+                    0,
+                )
+                .expect("Failed to create public for primary key"),
+                Some(&primary_key_auth),
+                None,
+                None,
+                None,
+            )
+            .expect("Failed to create primary handle")
+            .key_handle
+        });
+
+        context
+            .tr_set_auth(primary_key_handle.into(), &primary_key_auth)
+            .expect("Failed to set auth from primary key handle.");
+
+        let symmetric_key_object_attributes = ObjectAttributesBuilder::new()
+            .with_user_with_auth(true)
+            .with_sign_encrypt(true)
+            .with_decrypt(true)
+            .build()
+            .expect("Failed to create object attributes for symmetric key");
+
+        let symmetric_key_public = Tpm2BPublicBuilder::new()
+            .with_type(AlgorithmIdentifier::SymCipher.into()) // This is a flaw in the builder
+            .with_name_alg(AlgorithmIdentifier::Sha256.into()) // This is a flaw in the builder.
```

Comment:
What are the flaws here? The input type?

Reply:
Yes, it uses the raw TSS types.